src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java

```diff
@@ -530,14 +528,12 @@ private boolean hasEmptyLineAfter(DetailAST token) {
     private boolean hasEmptyLine(int startLine, int endLine) {
         // Initial value is false - blank line not found
         boolean result = false;
-        if (startLine <= endLine) {
```

Comment:
Condition was removed because the for loop does the same test before starting the loop.